setup.py

```diff
@@ -48,19 +52,26 @@ def run_tests(self):
     "f90nml",
     "pyam-iamc",
     "PyYAML",
+    "expectexception @ git+https://github.com/thedataincubator/expectexception.git@master",
+     "ipywidgets",
+     "appmode",
 ]
 
-extra_requirements = {"test": ["pytest", "pytest-cov", "codecov", "goodtables"]}
+extra_requirements = {
+    "tests": ["pytest>=4.0", "pytest-cov", "codecov", "goodtables", "notebook", "matplotlib", "nbval", "goodtables"],
+    "docs": ["sphinx>=1.4", "sphinx_rtd_theme", "sphinx-autodoc-typehints", "pydoc-markdown"],
+    "deploy": ["setuptools>=38.6.0", "twine>=1.11.0", "wheel>=0.31.0", "flake8", "black", "versioneer"],
```

Comment:
Maybe we should have a separate group `dev` or call the whole thing `dev` as only the first three are needed for PyPI.

Versioneer is not needed here i think, it's more a system tool that creates the necessary files in your project. Or is it needed to update `versioneer` from time to time?

Reply:
> Maybe we should have a separate group `dev` or call the whole thing `dev` as only the first three are needed for PyPI.

Ok.

> Or is it needed to update `versioneer` from time to time?

Nope I don't think so, have removed.